### ISSUE TITLE
cxxnet with the google net pretrained model

### ISSUE THREAD

**Issue by kaishijeng** [dmlc/cxxnet#172]
I download the pretrained google model, 0039.model and create a conf, test-goog.conf file (see below) to do a prediction, by using 
../../cxxnet ./test-goog.conf
There is an error message below with the above command:

finish initialization with 1 devices
ConvolutionLayer: LoadModel invalid model file

Is anything wrong with the test-goog.conf below:

pred = pred.txt
iter = image
  image_list = "pred.lst"
  image_root = "/home/hevc/src/cxxnet/images"
  image_mean = "mean_224.bin"
#no random crop and mirror in test
iter = end
task = pred
model_in = ./models/0039.model

netconfig = start
The rest is copied from Inception-BN.conf
Thanks

Kaishi

**Comment by antinucleon** [comment]
I just test on my machine with newest clone, there nothing wrong on my machine. 

model md5: ```9c96f20f9dfdc3a20ca73983e2851d34  0039.model```

**Comment by kaishijeng** [comment]
Antinucleon:

I check the downloaded model with md5sum and it also gives me the md5 value as you have.
9c96f20f9dfdc3a20ca73983e2851d34.

How do you verify the model with cxxnet?
I will try to use the  latest clone code and see whether the error will go away or not.


FC

**Comment by antinucleon** [comment]
I set conf to
```
eval = val
iter = imgbin
  image_list = /home/bing/Data/ImageNet/cxxnet/val/val.lst
  image_bin = /home/bing/Data/ImageNet/cxxnet/val/val.bin
  image_mean = "mean_224.bin"
iter = end
```
and I set ```model_in``` to check evaluation score.

**Comment by kaishijeng** [comment]
No luck with the latest git pull. Still give me the same error.

Kaishi

**Comment by kaishijeng** [comment]
Is this model only for GPU?
My system uses CPU.

Kaishi

**Comment by antinucleon** [comment]
There is no difference of GPU/CPU model in cxxnet. I just try to set ```dev=cpu```, it also works well.

**Comment by zhongwen** [comment]
Caused by inconsistent length of `long` type in 32-bit system. 

Fixed in 9df6af8. 

@kaishijeng Thx for reporting this.
